src/System.Net.NetworkInformation/src/System/Net/NetworkInformation/LinuxNetworkInterface.cs

```diff
@@ -68,15 +68,35 @@ private static LinuxNetworkInterface GetOrCreate(Dictionary<string, LinuxNetwork
             return lni;
         }
 
-        public override bool SupportsMulticast { get { return _supportsMulticast; } }
+        public override bool SupportsMulticast
+        {
+            get
+            {
+                if (_supportsMulticast.HasValue)
+                {
+                    return _speed.Value;
+                }
+                else
+                {
+                    throw new PlatformNotSupportedException(SR.net_InformationUnavailableOnPlatform);
+                }
+            }
+        }
 
-        private static bool GetSupportsMulticast(string name)
+        private static bool? GetSupportsMulticast(string name)
         {
             // /sys/class/net/<interface_name>/flags
             string path = Path.Combine(NetworkFiles.SysClassNetFolder, name, NetworkFiles.FlagsFileName);
-            Interop.LinuxNetDeviceFlags flags = (Interop.LinuxNetDeviceFlags)StringParsingHelpers.ParseRawHexFileAsInt(path);
 
-            return (flags & Interop.LinuxNetDeviceFlags.IFF_MULTICAST) == Interop.LinuxNetDeviceFlags.IFF_MULTICAST;
+            if (File.Exists(path))
```

Comment:
I don't know if the files in question could go away after the check and before you try to open it.  If it's possible, to be safe you'd want to wrap the body of the if in a try/catch for a FileNotFoundException.